public/js/src/module/connection.js

```diff
@@ -653,14 +653,15 @@ function getDataFile(url, languageMap) {
         })
         .catch((error) => {
             const errorMsg =
-                error.message ||
-                t('error.dataloadfailed', {
-                    url,
-                    // switch off escaping just for this known safe value
-                    interpolation: {
-                        escapeValue: false,
-                    },
-                });
+                !error.message || /fetch/.test(error.message)
+                    ? t('error.dataloadfailed', {
+                          filename: url.replace(/.*\//, ''),
+                          // switch off escaping just for this known safe value
+                          interpolation: {
+                              escapeValue: false,
+                          },
+                      })
+                    : error.message;
```

Comment:
This means that the first missing instance will be shown to the user, e.g. :

"Failed to load clinicaldata.xml."

I think that is likely more useful than a generic 'Failed to load external data'.